### PR TITLE
Bug functionality

### PR DESCRIPTION
**Pull request by faiza-ashfaq**

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -57,23 +57,22 @@ def update
     private
     def sign_up_params
```
Comment (Ume-Kalsoom) [inline app/controllers/users/registrations_controller.rb:58]: these 2 methods `sign_up_params` and `account_update_params` are exactly same, so we can use the same where needed

File: app/policies/project_policy.rb [hunk 2]
```diff
@@ -19,17 +21,11 @@ def new?
   end
 
   def update?
-    #byebug
+    # byebug
```
Comment (Ume-Kalsoom) [inline app/policies/project_policy.rb]: remove this line

File: app/controllers/bugs_controller.rb [hunk 3]
```diff
@@ -1,4 +1,91 @@
...
+    if @bug.update(bug_params)
+      flash[:notice] = 'Bug updated successfully'
+      redirect_to project_bugs_path(@project)
+    else
+      flash[:alert] = 'Bug failed to update'
+      render 'new'
+    end
+  end
+
+  def edit; end
```
Comment (Ume-Kalsoom) [inline app/controllers/bugs_controller.rb:44]: no authorization required here?

File: app/controllers/concerns/non_rest.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module NonRest
```
Comment (Ume-Kalsoom) [inline app/controllers/concerns/non_rest.rb:3]: rename to something like `BugStatusConcern`

File: app/services/application_service.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class ApplicationService
+  def self.call(*args, &block)
```
Comment (Ume-Kalsoom) [inline app/services/application_service.rb:4]: do we need this file?

File: app/views/bugs/_bug.html.erb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<tr>
+        <td><%= bug.title %></td>
```
Comment (Ume-Kalsoom) [inline app/views/bugs/_bug.html.erb:2]: indentation issues

File: app/services/application_service.rb (hunk 5 above)
Comment (faiza-ashfaq) [reply app/services/application_service.rb:4]: Yes, the other service inherits from this. This is not important but makes the syntax consistent if i add more service objects

**Review by Ume-Kalsoom — APPROVED** [review]

**Pull request merged**